src/arch/xtensa/configs/baytrail_defconfig

```diff
@@ -14,3 +14,11 @@ CONFIG_OPTIMIZE_FOR_SIZE=y
 CONFIG_HAVE_AGENT=n
 CONFIG_DEBUG_MEMORY_USAGE_SCAN=n
 CONFIG_TRACE_FILTERING_ADAPTIVE=n
+CONFIG_PCM_CONVERTER_FORMAT_S24LE=n
+CONFIG_PCM_CONVERTER_FORMAT_S32LE=n
+CONFIG_PCM_CONVERTER_FORMAT_FLOAT=n
+CONFIG_FORMAT_S24LE=n
+CONFIG_FORMAT_S32LE=n
```

Comment:
We may need to restrict this somehow alongside the pipeline i.e. running 24 bit internally PCM to DAI may work here.

Reply:
Why would we change this now? This is legacy stuff and unlikely to save much memory really.

Reply:
maybe it's the module adapter that's just too fat...

Reply:
More seriously @ranj063, this means that the module adapter will require more memory on ALL platforms, since volume is a common tools used across the board. That's really concerning to me.

Reply:
@plbossart the module adapter will require marginally more memory, yes. But I dont think it should be concerning enough. I will be back with some memory consumption stats soon. BUt the problem with BYT/CHT is that we're already too close to the max text size so any addition of code breaks them

Reply:
@ranj063 I remain unclear as to why would we need the module adapter for BYT/CHT? Things worked fine without it...

Reply:
with all due respect, this sounds like "when all you have is a hammer, everything looks like a nail"

I've seen this many times over the years, abstraction layers on top of modular interfaces breaking legacy platforms...It's hard to "think small' but that's how you will avoid losing cycles/memory. Trying to make BYT work is looking at the wrong side of the lens IMHO. Been there, done that, etc.

Reply:
@plbossart thanks for pushing back. I found a way to keep the original modular interface with a kconfig option. @lgirdwood we dont need this PR anymore.

Reply:
@ranj063 ok, if not needed any more then I can reduce the defconfig to remove float and keep the rest which are not used or needed on the other platforms.